### PR TITLE
Move OCSAuthAPI to AppFramework

### PR DESCRIPTION
**Pull request by rullzer**
- Convert class
- Convert routes
- Convert tests

CC: @LukasReschke @nickvergessen @MorrisJobke @icewind1991

### CODE REVIEW

**Comment by icewind1991** [comment]
:+1: looks good

File: apps/federation/lib/Controller/OCSAuthAPIController.php [hunk 1]
```diff
@@ -89,18 +91,21 @@ public function __construct(
 	}
 
 	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 * @PublicPage
```
Comment (LukasReschke) [inline apps/federation/lib/Controller/OCSAuthAPIController.php]: PublicPage implies @NoAdminRequired ;)

Comment (rullzer) [inline apps/federation/lib/Controller/OCSAuthAPIController.php]: Ah fair enough. But it doesn't hurt either.

Comment (rullzer) [inline apps/federation/lib/Controller/OCSAuthAPIController.php]: fixed

**Comment by MorrisJobke** [comment]
Tested and works (with existing and new federated shares) 👍

**Pull request merged**